_examples/custom/custom.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"fmt"
+	"gopkg.in/vinxi/metrics.v0"
+	"gopkg.in/vinxi/vinxi.v0"
+)
+
+const port = 3100
+
+func main() {
+	// Create a new vinxi proxy
+	vs := vinxi.NewServer(vinxi.ServerOptions{Port: port})
+
+	// Attach the metrics middleware
+	vs.Use(metrics.New(collector(collect)))
+
+	// Target server to forward
+	vs.Forward("http://httpbin.org")
+
+	fmt.Printf("Server listening on port: %d\n", port)
+	err := vs.Listen()
+	if err != nil {
+		fmt.Errorf("Error: %s\n", err)
+	}
+}
+
+// Simple stub collector
+type collector func(metrics.Reporter)
```

Comment:
Took me some time to understand how this example works. I think it may be easier to understand if you used a simple struct instead of a function.